### PR TITLE
svelte media bindings

### PR DESCRIPTION
**Pull request by jasonlyu123**
#500

### CODE REVIEW

**Comment by jasonlyu123** [comment]
The `readonly` modifier isn't needed. It's just for documentation purposes. And it could be easiler for future works if someday we can typecheck readonly props.

File: packages/svelte2tsx/svelte-jsx.d.ts [hunk 1]
```diff
@@ -716,6 +716,40 @@
...
+        readonly buffered?: SvelteMediaTimeRange[];
+        readonly played?: SvelteMediaTimeRange[];
+        readonly seekable?: SvelteMediaTimeRange[];
+        readonly seeking?: boolean;
+        readonly ended?: boolean;
+
+        /**
+         * the current playback time in the video, in seconds
+         */
+        currentTime?: number;
```
Comment (dummdidumm) [inline packages/svelte2tsx/svelte-jsx.d.ts:735]:
```suggestion
        currenttime?: number;
```

File: packages/svelte2tsx/svelte-jsx.d.ts [hunk 2]
```diff
@@ -716,6 +716,40 @@
...
+        /**
+         * how fast or slow to play the video, where 1 is 'normal'
+         */
+        playbackRate?: number;
```
Comment (dummdidumm) [inline packages/svelte2tsx/svelte-jsx.d.ts:739]:
```suggestion
        playbackrate?: number;
```

File: packages/svelte2tsx/svelte-jsx.d.ts [hunk 3]
```diff
@@ -716,6 +716,40 @@
...
+        paused?: boolean;
+        /**
+         * a value between 0 and 1
+         */
+        volume?: number;
+        muted?: boolean;
+    }
+
+    interface SvelteVideoProps extends SvelteMediaProps {
+        readonly videoWidth?: number;
```
Comment (dummdidumm) [inline packages/svelte2tsx/svelte-jsx.d.ts:749]:
```suggestion
        readonly videowidth?: number;
```

File: packages/svelte2tsx/svelte-jsx.d.ts [hunk 4]
```diff
@@ -716,6 +716,40 @@
...
+        readonly videoHeight?: number;
```
Comment (dummdidumm) [inline packages/svelte2tsx/svelte-jsx.d.ts:750]:
```suggestion
        readonly videoheight?: number;
```

**Review by dummdidumm — CHANGES_REQUESTED** [review]
Nice idea of not having it as part of the big `HTMLAttributes` list 👍  I suggested some lowercase changes because all props are lowercased by `svelte2tsx`. Also some properties which are on `HTMLAttributes`, like `defaultplaybackrate`, should be moved into this new interface I think.

**Comment by jasonlyu123** [comment]
I have to look into it. all lowercase would have typescript errors on two way binding but camel case won't have error in both props and two way binding.

**Comment by jasonlyu123** [comment]
two-way binding is not being transformed to all lowercase. Another problem is most of these camel case props are not HTML attributes but svelte props. The compiler would complain when you use all lower cases which means the autocomplete is wrong.

**Comment by dummdidumm** [comment]
Damn that's tricky ... About the autocompletion: We could try to turn of autocompletion for all JSX intrinsic elements, they should be shown anyway through the html language service. This would have the additional benefit of not everything showing up twice. If that works, you could just have two variants inside the JSX typings - one all lowercase for the attribute, one camelcased for the binding.

**Comment by dummdidumm** [comment]
I just noticed that the fix in #496 for #494 has the same problem (so #494 is not fixed yet). My bad.

**Comment by jasonlyu123** [comment]
What do you think about also transform two-way binding to lowercase?  We can let the svelte compiler validate the casing on the binding on the svelte-specific props.

**Comment by dummdidumm** [comment]
Good idea, I think this will work. It's transformed to a simple input property, so we can lowercase it like we do with the real input properties. But we should only do this for html elements, not component elements.

**Comment by jasonlyu123** [comment]
It seems like only binding is case-sensitive that's just put two variant as you suggest earlier

**Comment by jasonlyu123** [comment]
Some of it doesn't work when used as HTML attribute. It seems like they are svelte spefic props. So I removed the lowercase variant 
```svelte
<script>let playbackRate = 1</script>
<video playbackRate={1} /> <!--doesn't work -->
<video bind:playbackRate /> <!--work -->
```

**Review by dummdidumm — APPROVED** [review]
👍

**Pull request merged**